javascript/app-model-exposure/src/components/DefinitionForm.tsx

```diff
@@ -49,6 +47,30 @@ export const DefinitionForm: FC<{
 	const { ref, width } = useDimensions()
 	const headers = useHeaders(width)
 
+	const handlePrimary = useCallback(() => {
+		const primaryDefinition = definitions.find(
+			d => d.level === CausalityLevel.Primary && d.type === definitionType,
+		)
+		let definitionsCopy = [...definitions]
```

Comment:
since we're only mutating definitions when primaryDefinition changes, let's move the copy and setter into the `if` statement, and then use a `const newDefinitions = definitions.map()` statement.

Reply:
I refactored it to this: https://github.com/microsoft/showwhy/pull/354/files#diff-04017b7a9cbd8ed318b0fb4f9efc06f0f3e11dbb382fdbbc9152104f7fa59afe